File: Labtwo/PHYTHON/pr3.py
def funcnum(numb):
    # Проверка что все символы - цифры
    if not numb.isdigit():
        return
    
    # Проверка длины
    if len(numb) < 2:
        return
    
    num = int(numb)
    sum_d = 0
    product = 1
    
    print(f"{num}: ", end="")
    
    # Сумма цифр
    for i, c in enumerate(numb):
        d = int(c)
        sum_d += d
        print(c, end="")
        if i < len(numb) - 1:
            print("+", end="")
    print(f"={sum_d}, ", end="")
    
    # Произведение цифр
    for i, c in enumerate(numb):
        d = int(c)
        product *= d
        print(c, end="")
        if i < len(numb) - 1:
            print("*", end="")
    print(f"={product}, ", end="")
    
    # Модуль разности
    diff = abs(sum_d - product)
    print(f"|{sum_d}-{product}|={diff}")

print("Write your number")
input_str = input().strip()
for numb in input_str.split():
    funcnum(numb)

File: Labtwo/SWIFT/pr3.swift
import Foundation

func funcnum(_ numb: String) {
    // Проверка что все символы - цифры
    guard CharacterSet.decimalDigits.isSuperset(of: CharacterSet(charactersIn: numb)) else {
        return
    }
    
    // Проверка длины
    guard numb.count >= 2 else {
        return
    }
    
    let num = Int(numb)!
    var sum = 0
    var proizv = 1
    
    print("\(num): ", terminator: "")
    
    // Сумма цифр
    for (i, c) in numb.enumerated() {
        let d = Int(String(c))!
        sum += d
        print(c, terminator: "")
        if i < numb.count - 1 {
            print("+", terminator: "")
        }
    }
    print("=\(sum), ", terminator: "")
    
    // Произведение цифр
    for (i, c) in numb.enumerated() {
        let d = Int(String(c))!
        proizv *= d
        print(c, terminator: "")
        if i < numb.count - 1 {
            print("*", terminator: "")
        }
    }
    print("=\(proizv), ", terminator: "")
    
    // Модуль разности
    let diff = abs(sum - proizv)
    print("|\(sum)-\(proizv)|=\(diff)")
}

print("Write your number")
if let input = readLine() {
    for numb in input.components(separatedBy: .whitespaces) {
        funcnum(numb)
    }
}